src/main/java/org/mockito/internal/invocation/InvocationMatcher.java

```diff
@@ -134,24 +133,12 @@ private void captureVarargsPart(Invocation invocation) {
             return;
         }
 
-        Object[] rawArguments = invocation.getRawArguments();
-        int indexOfVarArg = rawArguments.length - 1;
+        Object[] arguments = invocation.getArguments();
 
-        Object varArg = rawArguments[indexOfVarArg];
-        if (varArg == null) {
-            return;
-        }
-        int varArgSize = Array.getLength(varArg);
-
-        if (matchers.size() == indexOfVarArg) {
-            return;
-        }
-
-        for (int i = 0; i < varArgSize; i++) {
-            int indexOfVarargMatcher = (i + indexOfVarArg > matchers.size() -1 ? indexOfVarArg : i + indexOfVarArg);
-            ArgumentMatcher m = matchers.get(indexOfVarargMatcher);
+        for (int i = 0; i < arguments.length; i++) {
+            ArgumentMatcher m = matchers.get(i > matchers.size() - 1 ? matchers.size() - 1 : i);
             if (m instanceof CapturesArguments) {
-                ((CapturesArguments) m).captureFrom(Array.get(varArg, i));
+                ((CapturesArguments) m).captureFrom(arguments[i]);
             }
         }
```

Comment:
A `null`-Argument should be captured but a `null`-array not, imho it is like passing an empty array. E.g:

`mock.varArgs(null,null,null)`  <- should capture 3x null, it is equivalent to 
`mock.varArgs(new String[]{null,null,null})` 
`mock.varArgs((String[]) null)` <- should capture nothing cause there is nothing to capture

What do you think?

---

Are you sure that `mock.varArgs((String[]) null)` will not throw a _NullPointerException_ with the current implementation? Do we have a JUnit-Test?